hunting-buddy.lic

```diff
@@ -284,7 +284,24 @@ class HuntingBuddy
         return true
       else
         @exit = nil
-        rooms_to_search = @hunting_zones.select { |zone_name, zone_rooms| zones_to_search.include?(zone_name) }.values.flatten.compact
+        rooms_to_search = []
+
+        # After grabbing the list of hunting zones specified in the user's yaml, we'll iterate
+        # through each and aggregate the room numbers from base-hunting.yaml.
+        # NOTE: Be sure to preserve the order we aggregate the room numbers to match the order
+        # of the :zone: list in the yaml. Users expect the :zone: list to be a priority order.
+        # If they list prereni first, they want to try there first.
+        zones_to_search.each do |zone|
+          rooms_to_search += @hunting_zones["#{zone}"] unless @hunting_zones["#{zone}"] == nil
+        end
```

Comment:
Thanks for getting this fix in 👍 

💡 For knowledge share, another way this could be done is with `.map`, `.flatten`, and `.compact`.
```ruby
rooms_to_search = zones_to_search.map { |zone| @hunting_zones[zone] }.flatten.compact
```

_What's happening:_
1. `.map` iterates the `zones_to_search` array and transforms it from a list of zone names to a list of list of room ids. For example, `['rats', 'nope', 'goblins'] ==> [ [1,2,3], nil, [4,5,6] ]`
2. `.flatten` then transforms the list of list of room ids into one giant list of room ids, in the same order the elements were in. For example, `[ [1,2,3], nil, [4,5,6] ] ==> [1,2,3,nil,4,5,6]`
3. `.compact` then removes any `nil` elements from the list, in case any of the zones from the player's yaml file didn't have a match in `@hunting_zones` map. For example, `[1,2,3,nil,4,5,6] ==> [1,2,3,4,5,6]`

Reply:
Thanks for the knowledge share. Took a bit but I got it worked out mentally.